tests/testsuite/fix.rs

```diff
@@ -29,7 +29,10 @@ fn do_not_fix_broken_builds() {
     p.cargo("fix --allow-no-vcs")
         .env("__CARGO_FIX_YOLO", "1")
         .with_status(101)
-        .with_stderr_contains("[ERROR] could not compile `foo` (lib) due to 1 previous error")
+        .with_stderr_line_without(
+            &["[ERROR] could not compile `foo` (lib) due to 1 previous error"],
+            &[],
+        )
```

Comment:
Before <https://github.com/rust-lang/rust/pull/120550>:

```
error[E0308]: mismatched types
 --> src/lib.rs:8:35
  |
8 |                     let _x: u32 = "a";
  |                             ---   ^^^ expected `u32`, found `&str`
  |                             |
  |                             expected due to this

For more information about this error, try `rustc --explain E0308`.
error: could not compile `foo` (lib test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `foo` (lib) due to 1 previous error
```

This is what we'll have after <https://github.com/rust-lang/rust/pull/120550>:

```
error[E0308]: mismatched types
Error:  --> src/lib.rs:8:35
  |
8 |                     let _x: u32 = "a";
  |                             ---   ^^^ expected `u32`, found `&str`
  |                             |
  |                             expected due to this

warning: variable does not need to be mutable
Warning:  --> src/lib.rs:3:25
  |
3 |                     let mut x = 3;
  |                         ----^
  |                         |
  |                         help: remove this `mut`
  |
  = note: `#[warn(unused_mut)]` on by default

For more information about this error, try `rustc --explain E0308`.
warning: `foo` (lib) generated 1 warning
error: could not compile `foo` (lib) due to 1 previous error; 1 warning emitted
warning: build failed, waiting for other jobs to finish...
warning: `foo` (lib test) generated 1 warning (1 duplicate)
error: could not compile `foo` (lib test) due to 1 previous error; 1 warning emitted
```  


I wonder if we could 


```suggestion
        .with_stderr_contains("[ERROR] could not compile `foo` (lib) due to 1 previous error[..]")
```

And have a TODO for updating this after <https://github.com/rust-lang/rust/pull/120550> landed. (though not required since it's not the point of this test)

Reply:
yep, that works.